ydb/library/yql/providers/common/codec/yql_codec.cpp

```diff
@@ -52,98 +52,98 @@ void WriteYsonValueImpl(TYsonResultWriter& writer, const NUdf::TUnboxedValuePod&
                 writer.OnStringScalar(value.AsStringRef());
                 return;
 
-            case NUdf::TDataType<NUdf::TDecimal>::Id: {
+            case NUdf::EDataSlot::Decimal: {
                 const auto params = static_cast<TDataDecimalType*>(type)->GetParams();
                 const auto str = NDecimal::ToString(value.GetInt128(), params.first, params.second);
                 const auto size = str ? std::strlen(str) : 0;
                 writer.OnUtf8StringScalar(TStringBuf(str, size));
                 return;
             }
-            case NUdf::TDataType<NUdf::TYson>::Id:
+            case NUdf::EDataSlot::Yson:
                 EncodeRestrictedYson(writer, value.AsStringRef());
                 return;
-            case NUdf::TDataType<NUdf::TDate>::Id:
+            case NUdf::EDataSlot::Date:
                 writer.OnUint64Scalar(value.Get<ui16>());
                 return;
-            case NUdf::TDataType<NUdf::TDatetime>::Id:
+            case NUdf::EDataSlot::Datetime:
                 writer.OnUint64Scalar(value.Get<ui32>());
                 return;
-            case NUdf::TDataType<NUdf::TTimestamp>::Id:
+            case NUdf::EDataSlot::Timestamp:
                 writer.OnUint64Scalar(value.Get<ui64>());
                 return;
-            case NUdf::TDataType<NUdf::TInterval>::Id:
+            case NUdf::EDataSlot::Interval:
                 writer.OnInt64Scalar(value.Get<i64>());
                 return;
-            case NUdf::TDataType<NUdf::TTzDate>::Id:
-            case NUdf::TDataType<NUdf::TTzDatetime>::Id:
-            case NUdf::TDataType<NUdf::TTzTimestamp>::Id:
-            case NUdf::TDataType<NUdf::TJsonDocument>::Id: {
+            case NUdf::EDataSlot::TzDate:
+            case NUdf::EDataSlot::TzDatetime:
+            case NUdf::EDataSlot::TzTimestamp:
+            case NUdf::EDataSlot::TzDate32:
+            case NUdf::EDataSlot::TzDatetime64:
+            case NUdf::EDataSlot::TzTimestamp64:
+            case NUdf::EDataSlot::JsonDocument: {
                 const NUdf::TUnboxedValue out(ValueToString(*dataType->GetDataSlot(), value));
                 writer.OnUtf8StringScalar(out.AsStringRef());
                 return;
             }
-            case NUdf::TDataType<NUdf::TDate32>::Id:
+            case NUdf::EDataSlot::Date32:
                 writer.OnInt64Scalar(value.Get<i32>());
                 return;
-            case NUdf::TDataType<NUdf::TDatetime64>::Id:
+            case NUdf::EDataSlot::Datetime64:
                 writer.OnInt64Scalar(value.Get<i64>());
                 return;
-            case NUdf::TDataType<NUdf::TTimestamp64>::Id:
+            case NUdf::EDataSlot::Timestamp64:
                 writer.OnInt64Scalar(value.Get<i64>());
                 return;
-            case NUdf::TDataType<NUdf::TInterval64>::Id:
+            case NUdf::EDataSlot::Interval64:
                 writer.OnInt64Scalar(value.Get<i64>());
                 return;
-
-            default:
```

Comment:
Nit: Even though all the existing types are handled above, kinda `Y_ENSURE`/`Y_UNREACHABLE` (or `Y_DEBUG_ABORT_UNLESS` at least) is still worth having. If this violates the existing guidelines, feel free to ignore.

Reply:
"default" is bad practice for an enum class, it prevents the compiler from checking completeness